### PR TITLE
[API] Add AddressStateTx

### PR DESCRIPTION
**Pull request by c4t-ag**
### Implementation of SetAddressState which issues an AddAddressStateTx

Parameter:
```
type SetAddressStateArgs struct {
	api.JSONSpendHeader

	Address string `json:"address"`
	State   uint8  `json:"state"`
	Remove  bool   `json:"remove"`
}
```

- api.JSONSpendHeader: Keystore Creds, Froms and changeAddr
- Address: The address to change state for (roles / kyc)
- State: The state to set / remove (#66)
- Remove: Remove (true) or add (false) the state

Returns:
```
type SetAddressStateReply struct {
	api.JSONTxID
}
```
- api.JSONTxID: The txID of the transaction

### Implementation of GetAddressStateTx which returns an AddAddressStateTx

Parameter:
```
type GetAddressStateTxArgs struct {
	SetAddressStateArgs

	Encoding formatting.Encoding `json:"encoding"`
}
```

- SetAddressStateArgs: see above
- Encoding: encoding format of the resulting tx string

Returns:
```
type GetAddressStateTxReply struct {
	Tx string `json:"tx"`
}
```

- Tx: encoded AddAddressStateTx

### CODE REVIEW

File: vms/platformvm/service_test.go [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // Copyright (C) 2019-2022, Ava Labs, Inc. All rights reserved.
 // See the file LICENSE for licensing terms.
 
-//go:generate mockgen -source cache_internal_state.go -destination mock_cache_internal_state.go -package platformvm -aux_files github.com/chain4travel/caminogo/vms/platformvm=cache_versioned_state.go,github.com/chain4travel/caminogo/vms/platformvm=cache_validator_state.go
+//go:generate mockgen -source cache_internal_state.go -destination mock_cache_internal_state.go -package platformvm -aux_files github.com/ava-labs/avalanchego/vms/platformvm=cache_versioned_state.go,github.com/ava-labs/avalanchego/vms/platformvm=cache_validator_state.go
```
Comment (evlekht) [inline vms/platformvm/service_test.go:14]: I think we should remove that completely and work only with sh script

File: vms/platformvm/camino_service.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	// The maximum consumption rate
+	MaxConsumptionRate json.Uint64 `json:"maxConsumptionRate"`
+	// The supply cap for the native token (AVAX)
+	SupplyCap json.Uint64 `json:"supplyCap"`
+	// The codec version used for serializing
+	CodecVersion json.Uint16 `json:"codecVersion"`
+}
+
+// GetMinStake returns the minimum staking amount in nAVAX.
+func (service *Service) GetConfiguration(_ *http.Request, _ *struct{}, reply *GetConfigurationReply) error {
```
Comment (evlekht) [inline vms/platformvm/camino_service.go:65]: Why move that from avax service?

File: utils/crypto/rsa2secp256k1.go [hunk 3]
```diff
@@ -75,3 +76,13 @@ func RecoverSecp256PublicKey(cert *x509.Certificate) ([]byte, error) {
 
 	return hashing.PubkeyBytesToAddress(sPubKeyBytes), nil
 }
+
+func FakePrivateKey(addr ids.ShortID) *PrivateKeySECP256K1R {
```
Comment (evlekht) [inline utils/crypto/rsa2secp256k1.go:80]: Doesn't look like it belongs to this file `rsa2secp256k1.go` judging by its name

File: vms/platformvm/service_test.go (hunk 1 above)
Comment (evlekht) [reply vms/platformvm/service_test.go:14]: That line is from us from pre-banff changes

File: vms/platformvm/camino_service.go (hunk 2 above)
Comment (c4t-ag) [reply vms/platformvm/camino_service.go:65]: It's ours

File: vms/platformvm/service_test.go (hunk 1 above)
Comment (c4t-ag) [reply vms/platformvm/service_test.go]: We'll remove it in a different cleanup PR

**Review by evlekht — APPROVED** [review]

**Pull request merged**